### PR TITLE
TIMX-235 Update ASpace EAD date parsing

### PR DESCRIPTION
**Pull request by ghukill**
#### What does this PR do?

This PR udpates the logic for how dates are parsed for ASpace EADs.

Nearly all (~96%) ASpace EADs have a `@normal` attribute in the `archdesc.unitdate` element that is normalized and predictable in form.  This value is used now, split on `/` for date ranges, to populate dates for the TIMDEX record.

Other changes:
  * dependencies updated
  * `ipython` added as `--dev` package

#### Helpful background context
 We receieved a Sentry error that a record had an invalid date range.  As described in the linked ticket, it became apparent this was related to a combination of date parsing and validation, and how it varies between Transmogrifier and OpenSearch environments.

It has been noted in the linked Jira tickets that the EAD unitdate field @normal attribute might contain a more normalized form of the date that could help with these kind of errors or ambiguity.

#### How can a reviewer manually see the effects of these changes?

```shell
pipenv run transform --source aspace --input-file s3://timdex-extract-dev-222053980223/aspace/aspace-2023-08-16-full-extracted-records-to-index.xml --output-file ./output/aspace_date_parsing.json
```

Observe the results from transforming a large number of records.  Virtually all records should have date + date range, with some having `kind` and/or `note` included.

Example:
```json
"dates": [
    {
      "kind": "creation",
      "range": {
        "gte": "1905",
        "lte": "2012"
      }
    }
  ]
```

#### What are the relevant tickets?
  * https://mitlibraries.atlassian.net/browse/TIMX-235
  * https://mitlibraries.atlassian.net/browse/TIMX-92

#### Developer
- [X] All new ENV is documented in README
- [X] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer
- [ ] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [ ] There are appropriate tests covering any new functionality
- [ ] The documentation has been updated or is unnecessary
- [ ] The changes have been verified
- [ ] New dependencies are appropriate or there were no changes

#### Includes new or updated dependencies?
YES

### CODE REVIEW

File: transmogrifier/sources/ead.py [hunk 1]
```diff
@@ -452,3 +424,59 @@ def parse_mixed_value(
         elif isinstance(item, Tag) and item.name not in skipped_elements:
             for child in item.children:
                 yield from cls.parse_mixed_value(child, skipped_elements)
+
+    def parse_dates(
```
Comment (ehanson8) [inline transmogrifier/sources/ead.py:428]: Good call pulling this into a new method, it's too involved not to be separated out from `get_optional_fields`

File: transmogrifier/sources/ead.py [hunk 2]
```diff
@@ -452,3 +424,59 @@ def parse_mixed_value(
...
+                        gte_date,
+                        lte_date,
+                        source_record_id,
+                    ):
+                        date_instance.range = timdex.Date_Range(
+                            gte=gte_date,
+                            lte=lte_date,
+                        )
+                else:
+                    date_str = gte_date  # arbitrarily take one
```
Comment (ehanson8) [inline transmogrifier/sources/ead.py:460]: Optional, probably doesn't need to be assigned to another var with the `# arbitarily take one` note, you could just use `gte` with that note

**Review by ehanson8 — APPROVED** [review]
One optional suggestion

File: transmogrifier/sources/ead.py (hunk 2 above)
Comment (ghukill) [reply transmogrifier/sources/ead.py:460]: I'd considered that, but given it's used twice below, figured one line / one comment would inform the lines below.  Unless feel strongly, inclined to leave as-is.

**Pull request merged**